lib/stats/stats-cluster.h

```diff
@@ -59,7 +59,11 @@ struct _StatsCounterGroup
 
 struct _StatsCounterGroupInit
 {
-  const gchar **counter_names;
+  union
+  {
+    const gchar **names;
+    const gchar *name;
+  } counter;
```

Comment:
misleading naming - `name` is only valid for `SingleStatsCluster`
`counter` is actually `names_by_type`.
So the options are
1) refactor the whole stats and add multiple StatsCounterGroup types
2) extend `stats_counter_group_` functions (`stats_counter_group_single_get_name`) instead of adding the union